### PR TITLE
Finished part of first and second part of CIS108-a2

### PR DESCRIPTION
**Pull request by jazheng1**

### CODE REVIEW

**Comment by jazheng1** [comment]
included avgScrabbleScoreFromFile

File: src/avgScrabbleScore.js [hunk 1]
```diff
@@ -8,11 +8,13 @@ let scrabbleScore = require('./scrabbleScore');
  */
 function avgScrabbleScore(words) {
   // This is your job. :)
+  return scrabbleScore(words) / words.length;
```
Comment (jfarmer) [inline src/avgScrabbleScore.js]: `scrabbleScore` is meant to take a string as input and `avgScrabbleScore` is meant to take an array, so this couldn't work.

If `words` is a string then `words.length` is _not_ the average score per word, but rather the average score per letter.

File: src/avgScrabbleScoreFromFile.js [hunk 2]
```diff
@@ -12,11 +12,20 @@ let avgScrabbleScore = require('./avgScrabbleScore');
  */
 function avgScrabbleScoreFromFile(fileName) {
   // This is your job. :)
+  let text = fs.readFileSync(`${fileName}`, 'utf-8');
```
Comment (jfarmer) [inline src/avgScrabbleScoreFromFile.js]: 1. No reason to do `${fileName}` if `fileName` is already a string. `fs.readFileSync(fileName, 'utf-8')` will work.
2. You can use `trimRight()` to remove trailing whitespace.

If you want to handle newline characters in a cross-platform way with `split`:

- https://www.google.com/search?q=node+cross+platform+newline+split
- https://stackoverflow.com/questions/21895233/how-in-node-to-split-string-by-newline-n/21895299

File: src/scrabbleScore.js [hunk 3]
```diff
@@ -25,13 +25,31 @@ let SCRABBLE_TILE_SCORES = {
 
 function scrabbleScore(word) {
   // This is your job. :)
+  let adjustedWord = word.toLowerCase();
+  let counter = 0;
+  let sum = 0;
+  let keys = Object.keys(SCRABBLE_TILE_SCORES);
+  let value = Object.values(SCRABBLE_TILE_SCORES);
+  while (counter !== adjustedWord.length) {
```
Comment (jfarmer) [inline src/scrabbleScore.js]: I don't understand the triply-nested loop here. Iterate over each character and add `SCRABBLE_TILE_SCORES[letter]` to the running total as you go. One loop, one iteration per character.

**Pull request closed without merging**